111-use-buildah-instead-of-kaniko.md

```diff
@@ -0,0 +1,236 @@
+
+## Proposal
+
+For the Connect Build feature on Kubernetes, we will use [Buildah](https://buildah.io/) as a replacement of Kaniko.
+Buildah is well-supported, widely used tool for building container images.
+It also appears that OpenShift’s Build API relies on Buildah, which makes Buildah a strong alternative to Kaniko, with the added assurance of long-term support and maintenance.
+Other than that, Buildah doesn't need any workaround in order to run it on Kubernetes rootlessly - the only thing that is needed during the build and push stages is to specify `--storage-driver=vfs` option.
+VFS (Virtual File System) driver ensures user-space implementation, it stores each layer as a full copy of the files, but without need of root permissions.
+The only trade-off is the build time (it's slower) and consumption of disk space.
+
+Even though we will run Buildah with these options, which should make it rootless, it still needs some capabilities to run without issues during the build phase.
+That means we cannot run the build Pod with restricted Pod Security profile, as these capabilities are dropped in these profiles.
+This is more described in the [Running Buildah in restricted environment](#running-buildah-in-restricted-environment) section of this proposal.
+
+We will use the official Buildah images on Quay.io - currently `quay.io/buildah/stable:v1.40.1` - and similarly to Kaniko executor image - we will pull the official image, tag it with the version and push it to the Strimzi repository on Quay.
+That's useful because of our versioning and we will have the image ready even if someone on Buildah side decides to remove it from their Quay repository.
+The full name of our Buildah image will be `quay.io/strimzi/buildah:OPERATOR_VERSION` - for `main` branch it will use the `latest` tag, otherwise it will be tagged by the operator version for the particular release.
+
+Users then can specify their own Buildah image using `STRIMZI_DEFAULT_BUILDAH_IMAGE` (similarly to what is possible today with Kaniko).
+In case that users will use Kaniko, the Buildah environment variable will be ignored, and vice versa.
+The users will be responsible for changing these environment variables based on the mode in which they run the Connect build - in case that they have customized Deployment files.
+
+The usage of Buildah will be gated behind feature gate called `UseConnectBuildWithBuildah` that will have following schedule:
+
+| Alpha (opt-in) | Beta (default-on) | GA     |
+|----------------|-------------------|--------|
+| 0.49.0         | 0.52.0            | 0.55.0 |
+
+The build of the Connect image will be done the same way as today, the main difference will be in the commands used in the build and push process.
+Kaniko executor did everything at once, which can be done by Buildah as well, but I rejected to have it done in one command because:
```

Comment:
```suggestion
Kaniko executor did everything at once, which can be done by Buildah as well, but I decided not to implement it as a single command because:
```